litmus-portal/frontend/src/pages/MonitoringDashboardPage/index.tsx

```diff
@@ -164,17 +464,231 @@ const DashboardPage: React.FC = () => {
+        (event) => event.showOnTable || chaosDataSet.queryIDs.includes(event.id)
+      ),
+      numOfWorkflows: chaosInformation.numberOfWorkflowsUnderConsideration,
+      firstLoad: !analyticsData?.ListWorkflow,
+    });
+    const existingEventIDs: string[] = prometheusQueryData.chaosEvents.map(
+      ({ id }) => id
+    );
+    const newEventIDs: string[] = chaosInformation.chaosEventList
+      .map(({ id }) => id)
+      .filter((id: string) => !existingEventIDs.includes(id));
+    if (newEventIDs.length) {
+      setEventsToShow({
+        eventsToShow: selectedDashboardInformation.selectionOverride
+          ? eventsToShow.eventsToShow
+          : eventsToShow.eventsToShow.concat(newEventIDs),
+        selectEvents: true,
+      });
+    }
+    chaosInformation = {
+      promQueries: [],
+      chaosQueryIDs: [],
+      chaosEventList: [],
+      numberOfWorkflowsUnderConsideration: 0,
+    };
+  };
+
+  useEffect(() => {
+    if (prometheusQueryData.firstLoad) {
+      refetch();
+      generateChaosQueries();
+      if (selectedDashboard.refreshRate !== 2147483647) {
+        dashboard.selectDashboard({
+          range: {
+            startDate: moment
+              .unix(Math.round(new Date().getTime() / 1000) - 1800)
+              .format(),
+            endDate: moment
+              .unix(Math.round(new Date().getTime() / 1000))
+              .format(),
+          },
+        });
+      }
+    }
+    if (!prometheusQueryData.firstLoad) {
+      // check and update range for the default relative time selection
+      // for user selections / check with entire range of relative time differences
+      // for absolute time no updates to the range therefore no data refresh / feature disabled
+      const endDate: number =
+        new Date(moment(selectedDashboard.range.endDate).format()).getTime() /
+        1000;
+      const now: number = Math.round(new Date().getTime() / 1000);
+      const diff: number = Math.abs(now - endDate);
+      const maxLim: number =
+        (selectedDashboard.refreshRate ?? 10000) / 1000 !== 0
+          ? (selectedDashboard.refreshRate ?? 10000) / 1000 + 4
+          : 14;
+      if (
+        diff >= 0 &&
+        diff <= maxLim &&
+        selectedDashboard.refreshRate !== 2147483647
+      ) {
+        const startDate: number =
+          new Date(
+            moment(selectedDashboard.range.startDate).format()
+          ).getTime() / 1000;
+        const interval: number = endDate - startDate;
+        dashboard.selectDashboard({
+          range: {
+            startDate: moment
+              .unix(Math.round(new Date().getTime() / 1000) - interval)
+              .format(),
+            endDate: moment
+              .unix(Math.round(new Date().getTime() / 1000))
+              .format(),
+          },
+        });
+      }
+      setTimeout(
+        () => {
+          refetch();
+          generateChaosQueries();
+        },
+        selectedDashboard.refreshRate !== 0
```

Comment:
The default refresh rate 10000 should be in some constant and not hardcoded

Reply:
will create a constants file, although it might increase the file length again.